File: frontend/src/types/project.ts
export type ProjectStatus = "draft" | "study" | "quote" | "signed" | "installed";

export interface Project {
  id: string;
  user_id: string;
  client_id: string | null;
  name: string;
  address: string | null;
  lat: number;
  lon: number;
  status: ProjectStatus;
  notes: string | null;
  created_at: string;
  updated_at: string;
  client_name?: string;
  panel_count?: number;
}

export interface CreateProject {
  name: string;
  address?: string;
  lat: number;
  lon: number;
  client_id?: string;
  notes?: string;
}

export interface UpdateProject {
  name?: string;
  address?: string;
  lat?: number;
  lon?: number;
  status?: ProjectStatus;
  client_id?: string | null;
  notes?: string;
}

File: backend/app/schemas/project.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ProjectCreate(BaseModel):
    name: str
    client_id: uuid.UUID | None = None
    address: str | None = None
    lat: float
    lon: float
    status: str = "draft"
    notes: str | None = None


class ProjectUpdate(BaseModel):
    name: str | None = None
    client_id: uuid.UUID | None = None
    address: str | None = None
    lat: float | None = None
    lon: float | None = None
    status: str | None = None
    notes: str | None = None


class ProjectRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    client_id: uuid.UUID | None = None
    name: str
    address: str | None = None
    lat: float
    lon: float
    status: str
    notes: str | None = None
    created_at: datetime
    updated_at: datetime
    client_name: str | None = None
    panel_count: int = 0
